ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizerGroupMetrics.java

```diff
@@ -0,0 +1,192 @@
+import static com.netease.arctic.server.optimizing.TaskRuntime.Status.SCHEDULED;
+
+import com.netease.arctic.ams.api.metrics.Gauge;
+import com.netease.arctic.ams.api.metrics.Metric;
+import com.netease.arctic.ams.api.metrics.MetricDefine;
+import com.netease.arctic.ams.api.metrics.MetricKey;
+import com.netease.arctic.server.metrics.MetricRegistry;
+import com.netease.arctic.server.resource.OptimizerInstance;
+import org.apache.iceberg.relocated.com.google.common.annotations.VisibleForTesting;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class OptimizerGroupMetrics {
+
+  @VisibleForTesting protected static final String GROUP_TAG = "group";
+  public static final MetricDefine OPTIMIZER_GROUP_PENDING_TASKS =
+      defineGauge("optimizer_group_pending_tasks")
+          .withDescription("Number of pending tasks in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_EXECUTING_TASKS =
+      defineGauge("optimizer_group_executing_tasks")
+          .withDescription("Number of executing tasks in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_PLANING_TABLES =
+      defineGauge("optimizer_group_planing_tables")
+          .withDescription("Number of planing tables in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_PENDING_TABLES =
+      defineGauge("optimizer_group_pending_tables")
+          .withDescription("Number of pending tables in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_EXECUTING_TABLES =
+      defineGauge("optimizer_group_executing_tables")
+          .withDescription("Number of executing tables in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_OPTIMIZERS =
+      defineGauge("optimizer_group_optimizers")
+          .withDescription("Number of optimizer instances in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_MEMORY_BYTES_ALLOCATED =
+      defineGauge("optimizer_group_memory_bytes_allocated")
+          .withDescription("Memory bytes allocated in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  public static final MetricDefine OPTIMIZER_GROUP_THREADS =
+      defineGauge("optimizer_group_threads")
+          .withDescription("Number of total threads in optimizer group")
+          .withTags(GROUP_TAG)
+          .build();
+
+  private final String groupName;
+  @VisibleForTesting private final MetricRegistry registry;
```

Comment:
`@VisibleForTesting` seems to be unused.

Reply:
done